main.go

```diff
@@ -185,27 +193,89 @@ func printUsage() {
 	flag.PrintDefaults()
 }
 
+func getBuildInfo() *debug.BuildInfo {
+	bi, ok := debug.ReadBuildInfo()
+	if !ok {
+		return nil
+	}
+	return bi
+}
+
+func getMyModuleInfo(bi *debug.BuildInfo) (*debug.Module, error) {
+	if bi == nil {
+		return nil, fmt.Errorf("no build info available")
+	}
+	if bi.Main.Path != "" {
+		return &bi.Main, nil
+	}
+	for _, m := range bi.Deps {
+		if strings.HasPrefix(m.Path, "github.com/odenio/goimports-reviser") {
+			return m, nil
+		}
+	}
+	return nil, fmt.Errorf("weird: no matching dependency found in build info")
+}
+
 func printVersion() {
+	if Tag != "" {
+		fmt.Printf(
+			"version: %s\nbuilt with: %s\ntag: %s\ncommit: %s\nsource: %s\n",
+			strings.TrimPrefix(Tag, "v"),
+			GoVersion,
+			Tag,
+			Commit,
+			SourceURL,
+		)
+		return
+	}
+	bi := getBuildInfo()
+	myModule, err := getMyModuleInfo(bi)
+	if err != nil {
+		log.Fatalf("failed to get my module info: %s", err)
+	}
 	fmt.Printf(
-		"version: %s\nbuild with: %s\ntag: %s\ncommit: %s\nsource: %s\n",
-		strings.TrimPrefix(Tag, "v"),
-		GoVersion,
-		Tag,
-		Commit,
-		SourceURL,
+		"version: %s\nbuilt with: %s\ntag: %s\ncommit: %s\nsource: %s\n",
+		strings.TrimPrefix(myModule.Version, "v"),
+		bi.GoVersion,
+		myModule.Version,
+		"n/a",
+		myModule.Path,
 	)
 }
 
+func printVersionOnly() {
+	if Tag != "" {
+		fmt.Println(strings.TrimPrefix(Tag, "v"))
+		return
+	}
+	bi := getBuildInfo()
+	myModule, err := getMyModuleInfo(bi)
+	if err != nil {
+		log.Fatalf("failed to get my module info: %s", err)
+	}
+	fmt.Println(strings.TrimPrefix(myModule.Version, "v"))
+}
+
 func main() {
 	deprecatedMessagesCh := make(chan string, 10)
 	flag.Parse()
 
 	if shouldShowVersion != nil && *shouldShowVersion {
+		if shouldShowVersionOnly != nil && *shouldShowVersionOnly {
+			printVersionOnly()
+			return
+		}
 		printVersion()
 		return
 	}
 
+	if shouldShowVersionOnly != nil && *shouldShowVersionOnly {
```

Comment:
This `if` block is repeated. Once when in the if, one after. So the if is always performed.

If you move this `if` block before the previous one, you won't have to do the same `if` in the other block

So you will have only

```go
         if shouldShowVersionOnly != nil && *shouldShowVersionOnly {
                  printVersionOnly()
                  return 
         }

         if shouldShowVersion != nil && *shouldShowVersion { 
                  printVersion() 
                  return 
          }
```

Reply:
Oops, you're quite correct.  Updated.